qmrherm_module.F90

```diff
@@ -41,6 +41,7 @@ subroutine qmrherm(Phi, x, res, itercg, am, imass, anum, aden, ndiagq, iflag, us
     logical, intent(in), optional :: use_sp
     integer, intent(out) :: itercg
     integer, intent(out), optional :: cg_returns(ndiagq)
+    logical :: use_sp_flags_included = use_sp
```

Comment:
This will potentially fail if `use_sp` isn't present. I'd leave it uninitialised.